### PR TITLE
Undertow mod_proxy updates

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd [hunk 1]
```diff
@@ -535,5 +535,6 @@
         <xs:attribute name="request-queue-size" type="xs:int" use="optional" />
         <xs:attribute name="security-realm" type="xs:string" use="optional" />
         <xs:attribute name="use-alias" type="xs:string" use="optional" default="false" />
+        <xs:attribute name="enable-http2" type="xs:string" use="optional" default="false" />
```
Comment (rhusar) [inline undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd]: I think this was intended to be of type="xs:boolean" and also for the above use-alias.

Comment (stuartwdouglas) [inline undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd]: It takes an expression to technically is can be a string like ${host.http2}

Comment (rhusar) [inline undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd]: Ah ok, numerous other subsystems actually use the "right" type in XSD, but then use of expressions fail validation of course, rendering those XSD useless. So your way is better.

Comment (stuartwdouglas) [inline undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd]: I don't think we actually have a consistent guideline on how to handle this

**Comment by stuartwdouglas** [comment]
Retest this please

File: undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd (hunk 1 above)
Comment (bstansberry) [inline undertow/src/main/resources/schema/wildfly-undertow_3_0.xsd]: AFAIK the norm is to use the correct type and ignore the expression aspect. Most attributes support expressions, and I don't think being able to validate a document when everything is just xs:string is particularly meaningful. It doesn't prove much beyond valid structure, and unless every attribute with an expression follows this convention the document won't validate anyway.

I'm not really commenting on this PR here, as it looks like this xsd already uses xs:string everywhere. Just joining a conversation. :)

**Pull request merged**